paimon-common/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -1452,6 +1452,13 @@ public class CoreOptions implements Serializable {
                             "For DELETE manifest entry in manifest file, drop stats to reduce memory and storage."
                                     + " Default value is false only for compatibility of old reader.");
 
+    public static final ConfigOption<Boolean> DATA_FILE_THIN_MODE =
+            key("data-file.thin-mode")
+                    .booleanType()
+                    .defaultValue(true)
```

Comment:
Default true? Many other projects have integrated Paimon readers of older versions. If we produce thin data file by default, these old version readers won't be able to read Paimon files.

Reply:
I think 1.0 is a good time to make breaking changes which will be more difficult later.

Reply:
> I think 1.0 is a good time to make breaking changes which will be more difficult later.

Agree !

Reply:
In my side 1.0 is a very important version, many projects want to upgrade from 0.7/0.8/0.9 to 1.0, in default way need 
 to give these user a more easy way to upgrade their version.  At this version cut is not very suitable.

Reply:
We will change the default behavior in 1.x, maybe 1.1 or 1.2.

The point is that we need to make sure the read engines support file reading. They must upgrade the paimon lib to 1.0+.

Reply:
This is not break change if every engines use 1.0+ Paimon.

Reply:
Maybe we can add version metadata information in the paimon table to handle version compatibility issues.

Reply:
In later versions we will always encounter version compatibility issues.

Reply:
OK, I will change this to false.